### ISSUE TITLE
KeyError: 'content-disposition'

### ISSUE THREAD

**Issue by alexandrestevens** [Flimm/backup-all-my-flickr-photos#9]
Hello,
It works fine for the 25 first pics then i get this

```
-10649600% of -1 bytesTraceback (most recent call last):
  File "/usr/local/bin/backup-all-my-flickr-photos", line 235, in <module>
    main()
  File "/usr/local/bin/backup-all-my-flickr-photos", line 45, in main
    filename = automatic_retry(lambda: download_item(photo, download_dir))
  File "/usr/local/bin/backup-all-my-flickr-photos", line 221, in automatic_retry
    return function()
  File "/usr/local/bin/backup-all-my-flickr-photos", line 45, in <lambda>
    filename = automatic_retry(lambda: download_item(photo, download_dir))
  File "/usr/local/bin/backup-all-my-flickr-photos", line 152, in download_item
    extension = guess_extension(url, headers)
  File "/usr/local/bin/backup-all-my-flickr-photos", line 194, in guess_extension
    if headers['Content-Disposition']:
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/rfc822.py", line 393, in __getitem__
    return self.dict[name.lower()]
KeyError: 'content-disposition'
```

Can you help ?

Best
Alex

**Comment by Flimm** [comment]
@alexandrestevens Why did you close this? Do you no longer get this error?

**Comment by alexandrestevens** [comment]
@Flimm no... The solution i found is to download manually all videos and delete them from Flickr... and then restart the script...

**Comment by mpepping** [comment]
There seems to be an issue with downloading some of the videos indeed. The trace below shows that some MP4 files are saved correctly .. but then it borks on another (presumably) video-file:

```
Authorization saved to /.config/backup-all-my-flickr-photos/account_auth
Item 0001/7335 entitled
100% of 6.51 MB
Done: /backup/29894981126.mp4
Item 0002/7335 entitled
100% of 5.2 MB
Done: /backup/29894981066.mp4
Item 0003/7335 entitled
100% of 3.78 MB
Done: /backup/29815279082.mp4
Item 0004/7335 entitled
100% of 4.65 MB
Done: /backup/29814555942.mp4
Item 0005/7335 entitled
-10649600% of -1 bytesTraceback (most recent call last):
  File "/usr/local/bin/backup-all-my-flickr-photos", line 4, in <module>
    __import__('pkg_resources').run_script('backup-all-my-flickr-photos==0.7', 'backup-all-my-flickr-photos')
  File "/usr/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 658, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/local/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1445, in run_script
    exec(script_code, namespace, namespace)
  File "/usr/local/lib/python2.7/site-packages/backup_all_my_flickr_photos-0.7-py2.7.egg/EGG-INFO/scripts/backup-all-my-flickr-photos", line 239, in <module>

  File "/usr/local/lib/python2.7/site-packages/backup_all_my_flickr_photos-0.7-py2.7.egg/EGG-INFO/scripts/backup-all-my-flickr-photos", line 49, in main

  File "/usr/local/lib/python2.7/site-packages/backup_all_my_flickr_photos-0.7-py2.7.egg/EGG-INFO/scripts/backup-all-my-flickr-photos", line 225, in automatic_retry

  File "/usr/local/lib/python2.7/site-packages/backup_all_my_flickr_photos-0.7-py2.7.egg/EGG-INFO/scripts/backup-all-my-flickr-photos", line 49, in <lambda>

  File "/usr/local/lib/python2.7/site-packages/backup_all_my_flickr_photos-0.7-py2.7.egg/EGG-INFO/scripts/backup-all-my-flickr-photos", line 156, in download_item

  File "/usr/local/lib/python2.7/site-packages/backup_all_my_flickr_photos-0.7-py2.7.egg/EGG-INFO/scripts/backup-all-my-flickr-photos", line 198, in guess_extension

  File "/usr/local/lib/python2.7/rfc822.py", line 393, in __getitem__
    return self.dict[name.lower()]
KeyError: 'content-disposition'

```

The affected file was also a (h264) MP4 file, recorded with the MSQRD App (adds filters and animations). Excluding this video resolved the issue.  Don't know if it is of any help, but this is the `ffprobe` output on the affected file:

```
ffprobe ~/Downloads/29301581183.mp4 
ffprobe version 3.4.2 Copyright (c) 2007-2018 the FFmpeg developers
  built with Apple LLVM version 9.0.0 (clang-900.0.39.2)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/3.4.2 --enable-shared --enable-pthreads --enable-version3 --enable-hardcoded-tables --enable-avresample --cc=clang --host-cflags= --host-ldflags= --disable-jack --enable-gpl --enable-libmp3lame --enable-libx264 --enable-libxvid --enable-opencl --enable-videotoolbox --disable-lzma
  libavutil      55. 78.100 / 55. 78.100
  libavcodec     57.107.100 / 57.107.100
  libavformat    57. 83.100 / 57. 83.100
  libavdevice    57. 10.100 / 57. 10.100
  libavfilter     6.107.100 /  6.107.100
  libavresample   3.  7.  0 /  3.  7.  0
  libswscale      4.  8.100 /  4.  8.100
  libswresample   2.  9.100 /  2.  9.100
  libpostproc    54.  7.100 / 54.  7.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/Users/---/Downloads/29301581183.mp4':
  Metadata:
    major_brand     : mp42
    minor_version   : 1
    compatible_brands: mp41mp42isom
    creation_time   : 2016-06-24T17:42:14.000000Z
  Duration: 00:00:03.53, start: 0.000000, bitrate: 8471 kb/s
    Stream #0:0(und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 220 kb/s (default)
    Metadata:
      creation_time   : 2016-06-24T17:42:14.000000Z
      handler_name    : Core Media Audio
    Stream #0:1(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p(tv, bt709), 750x750, 8246 kb/s, 60 fps, 60 tbr, 600 tbn, 1200 tbc (default)
    Metadata:
      creation_time   : 2016-06-24T17:42:14.000000Z
      handler_name    : Core Media Video
```


.. Thanks for the scripts btw! 😊

**Comment by Flimm** [comment]
The new release, version 0.8, should fix this, thanks to the pull request #5.
